daemon/src/to_sse_event.rs

```diff
@@ -109,3 +111,66 @@ impl ToSseEvent for Option<Quote> {
         Event::json(self).event("quote")
     }
 }
+
+#[derive(Debug, Clone, Copy)]
+pub struct Heartbeat(time::OffsetDateTime);
+
+impl Heartbeat {
+    pub fn new() -> Self {
+        Self(time::OffsetDateTime::now_utc())
+    }
+}
+
+impl Default for Heartbeat {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl serde::Serialize for Heartbeat {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut heartbeat = serializer.serialize_struct("Heartbeat", 2)?;
+        heartbeat.serialize_field("event", "heartbeat")?;
+
+        let format = format_description!("[year]-[month]-[day]T[hour]:[minute]:[second]");
+        let timestamp = self.0.format(format).expect("should always format");
+        heartbeat.serialize_field("timestamp", &timestamp)?;
+
+        heartbeat.end()
+    }
+}
```

Comment:
Why this manual impl? I don't think you need to manually set the `event` field? We don't do this for any other structs?